### PR TITLE
feat: Toolbar Header

### PR DESCRIPTION
**Pull request by rodrigolabs**
## Description

This PR creates an example of how to build a Header with Toolbar using basically NavBar, Icons, Buttons, etc..

## Preview

image

image

## Deploy Preview

https://deploy-preview-1612--ohif.netlify.com/

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "test:e2e:dist": "lerna run test:e2e:dist --stream",
     "test:e2e:serve": "lerna run test:e2e:serve --stream",
     "see-changed": "lerna changed",
+    "docs:preview": "lerna run docs:preview --stream",
```
Comment (rodrigolabs) [inline package.json:38]: Created this script to run the documentation from the application root folder.

File: platform/ui/src/components/IconButton/IconButton.jsx [hunk 2]
```diff
@@ -87,15 +87,16 @@ const IconButton = ({
   disabled = defaults.disabled,
   type = defaults.type,
   fullWidth = defaults.fullWidth,
+  onClick,
```
Comment (rodrigolabs) [inline platform/ui/src/components/IconButton/IconButton.jsx:90]: Exposing `onClick` prop.

File: platform/ui/tailwind.config.js [hunk 3]
```diff
@@ -435,9 +436,10 @@ module.exports = {
     },
     padding: (theme) => theme('spacing'),
     placeholderColor: (theme) => theme('colors'),
-    stroke: {
+    stroke: (theme) => ({
+      ...theme('colors'),
```
Comment (rodrigolabs) [inline platform/ui/tailwind.config.js:440]: cc @galelis here's the setting that will allow us to use `stroke-primary-main` to colorize SVG strokes. With that, we will be able to set different colors for fill and stroke. :)

File: platform/ui/src/gatsby-theme-docz/tailwind.css [hunk 4]
```diff
@@ -22,3 +22,14 @@
 .ohif-scrollbar::-webkit-scrollbar-thumb:window-inactive {
   @apply bg-primary-main;
 }
+
+/* TOOLTIP WORKAROUND FOR ARROW UP */
+.tooltip::before {
```
Comment (galelis) [inline platform/ui/src/gatsby-theme-docz/tailwind.css:27]: Is this going to be used on all tooltips? maybe worth adding a custom class for this scenario.

File: platform/ui/src/components/Icon/getIcon.jsx [hunk 5]
```diff
@@ -21,11 +22,20 @@ import sorting from './../../assets/icons/sorting.svg';
 import sortingActiveDown from './../../assets/icons/sorting-active-down.svg';
 import sortingActiveUp from './../../assets/icons/sorting-active-up.svg';
 
+/** Tools */
+import toolZoom from './../../assets/icons/tool-zoom.svg';
+import toolCapture from './../../assets/icons/tool-capture.svg';
+import toolLayout from './../../assets/icons/tool-layout.svg';
+import toolMore from './../../assets/icons/tool-more-menu.svg';
+import toolMove from './../../assets/icons/tool-move.svg';
+import toolWindow from './../../assets/icons/tool-window-level.svg';
```
Comment (dannyrb) [inline platform/ui/src/components/Icon/getIcon.jsx:31]: I can't think of better names at the moment. I think the `tool` prefix works for now 👍

File: platform/ui/src/components/IconButton/IconButton.jsx [hunk 6]
```diff
@@ -87,15 +87,16 @@ const IconButton = ({
...
   ...rest
 }) => {
   const buttonElement = useRef(null);
 
   const handleOnClick = (e) => {
     buttonElement.current.blur();
-    if (rest.onClick) {
-      rest.onClick(e);
+    if (onClick) {
+      onClick(e);
```
Comment (dannyrb) [inline platform/ui/src/components/IconButton/IconButton.jsx:99]: I sometimes see a default `no operation` function for things like this, so we don't have to litter in checks:

```onClick: () => {}```

File: platform/ui/src/gatsby-theme-docz/tailwind.css [hunk 7]
```diff
@@ -22,3 +22,14 @@
...
+  @apply bg-primary-dark absolute z-10;
+  content: '';
+  width: 14px;
+  height: 1px;
+  top: -1px;
+  left: 50%;
+  transform: translateX(-50%);
+}
```
Comment (dannyrb) [inline platform/ui/src/gatsby-theme-docz/tailwind.css:35]: This is the gatsby theme or our theme?

Comment (rodrigolabs) [reply platform/ui/src/gatsby-theme-docz/tailwind.css:35]: yeah, some very specific custom styles are here. Better split?

Comment (dannyrb) [reply platform/ui/src/gatsby-theme-docz/tailwind.css:35]: Eventually. Fine to punt for now.

File: platform/ui/src/components/IconButton/IconButton.jsx (hunk 6 above)
Comment (rodrigolabs) [reply platform/ui/src/components/IconButton/IconButton.jsx]: I've fixed the `defaultProps` definition for this component and created a default method for `onClick`. Thanks!

File: platform/ui/src/gatsby-theme-docz/tailwind.css (hunk 4 above)
Comment (rodrigolabs) [reply platform/ui/src/gatsby-theme-docz/tailwind.css:27]: Yes! I was only thinking if we have in the future tooltip position like `up`, `left`, `bottom`...
But for now, we have only this kind of Tooltip.

File: platform/ui/src/gatsby-theme-docz/tailwind.css (hunk 7 above)
Comment (rodrigolabs) [reply platform/ui/src/gatsby-theme-docz/tailwind.css]: I've created a separate CSS to use when we need to create custom classes for the components.

**Pull request merged**